presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java

```diff
@@ -171,16 +171,34 @@ public TestingPrestoServer(List<Module> additionalModules)
         this(true, ImmutableMap.of(), null, null, new SqlParserOptions(), additionalModules);
     }
 
+    public TestingPrestoServer(boolean coordinator,
+                               Map<String, String> properties,
+                               String environment,
+                               URI discoveryUri,
+                               SqlParserOptions parserOptions,
+                               List<Module> additionalModules)
+            throws Exception
+    {
+        this(coordinator, properties, environment, discoveryUri, parserOptions, additionalModules, Optional.empty());
+    }
+
     public TestingPrestoServer(boolean coordinator,
             Map<String, String> properties,
             String environment,
             URI discoveryUri,
             SqlParserOptions parserOptions,
-            List<Module> additionalModules)
+            List<Module> additionalModules,
+            Optional<Path> baseDataDirPath)
             throws Exception
     {
         this.coordinator = coordinator;
-        baseDataDir = Files.createTempDirectory("PrestoTest");
+
+        if (!baseDataDirPath.isPresent()) {
```

Comment:
Use the `Optional` `map` and `orElse` apis instead of an if/else block

Reply:
@dain createTempDirectory("PrestoTest") throws IOException, and Optional.orElseGet() takes a Supplier<? extends T> parameter which doesn't support throwing checked exceptions. To get around it we need to provide a lambda wrapper of Supplier but then it loses the succinctness of using lambda here. Since we also need to remember isBaseDataDirUserSupplied, I reverted back to the original way as I think it's easier to read and understand. Please let me know if there are other better ways.